pkg/broker/ingress/handler.go

```diff
@@ -116,15 +113,13 @@ func (h *Handler) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Re
 	if len(pieces) != 3 {
 		msg := fmt.Sprintf("Malformed request path. want: '/<ns>/<broker>'; got: %v..", request.URL.Path)
 		h.logger.Info(msg)
-		response.WriteHeader(nethttp.StatusNotFound)
-		response.Write([]byte(msg))
+		nethttp.Error(response, msg, nethttp.StatusNotFound)
 		return
 	}
 	ns, broker := pieces[1], pieces[2]
 	event, msg, statusCode := h.toEvent(request)
 	if event == nil {
-		response.WriteHeader(statusCode)
-		response.Write([]byte(msg))
+		nethttp.Error(response, msg, statusCode)
 		return
 	}
```

Comment:
nit: it seems the `toEvent` func will only return bad request status code. Maybe just check err and return bad request here?

Reply:
`toEvent` can return different bad status codes, so we need it to return statusCode instead of just an error.

Reply:
Actually I am wrong :) Let me refactor it